pkgs/development/tools/rust/cargo-insta/default.nix

```diff
@@ -0,0 +1,34 @@
+{ stdenv, lib, rustPlatform, fetchFromGitHub, runCommand }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "cargo-insta";
+  version = "0.16.0";
+
+  src =
+    let
+      source = fetchFromGitHub {
+        owner = "mitsuhiko";
+        repo = "insta";
+        rev = version;
+        sha256 = "1lcbdzh139lhmpz3pyik8nbgrbfc42z9ydz2hkg2lzjdpfdsz3ag";
+      };
+    in
+    runCommand "cargo-insta-source" {} ''
```

Comment:
Even though it entails a diff for the whole lock file, it is a bit nicer to use `cargoPatches`, since it keeps the derivation more canonical.